### PR TITLE
Update auth.ts

### LINKED ISSUE #19879: Incorrect function is used in 'keycloak-admin-client' library in getToken function
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

adapter/javascript

### Describe the bug

It looks like it's just a simple mistake - 'atob' function is used instead of 'btoa'

### Version

2022.9

### Expected behavior

'btoa' function should be used

### Actual behavior

'atob' function is used

### How to Reproduce?

just run getToken function

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by hmica**
Correct getToken error

This fix a regression on migration from Axios To FetchAPI Credential header must be encoded not decoded.

This fix #19879

### CODE REVIEW

File: js/libs/keycloak-admin-client/src/utils/auth.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ export const getToken = async (settings: Settings): Promise<TokenResponse> => {
   if (credentials.clientSecret) {
     headers.set(
       "Authorization",
-      atob(credentials.clientId + ":" + credentials.clientSecret)
+      "Basic " + btoa(credentials.clientId + ":" + credentials.clientSecret)
```
Comment (jonkoops) [inline js/libs/keycloak-admin-client/src/utils/auth.ts:82]: Can we use a template literal here?
```suggestion
      `Basic ${btoa(`${credentials.clientId}:${credentials.clientSecret}`)`
```

**Review by jonkoops — CHANGES_REQUESTED** [review]
Perhaps it would be good to add some tests for this as well to prevent regressions in the future.

File: js/libs/keycloak-admin-client/src/utils/auth.ts (hunk 1 above)
Comment (hmica) [reply js/libs/keycloak-admin-client/src/utils/auth.ts]: yes, done !

**Comment by hmica** [comment]
Hi, I agree that's a good idea to have more test but I, not familiar with the project architecture, I try to run current test but they all failled, I suppose I need to have a Keycloak instance running.

File: js/libs/keycloak-admin-client/src/utils/auth.ts [hunk 2]
```diff
@@ -77,10 +77,8 @@ export const getToken = async (settings: Settings): Promise<TokenResponse> => {
   const headers = new Headers(options.headers);
 
   if (credentials.clientSecret) {
-    headers.set(
-      "Authorization",
-      atob(credentials.clientId + ":" + credentials.clientSecret)
-    );
+    const authHeaderValue = `Basic ${btoa(`${credentials.clientId}:${credentials.clientSecret}`)}`;
+    headers.set("Authorization", authHeaderValue);
```
Comment (jonkoops) [inline js/libs/keycloak-admin-client/src/utils/auth.ts:81]: I see no need to put this in a variable if it's only going to get used right after.
```suggestion
    headers.set("Authorization", `Basic ${btoa(`${credentials.clientId}:${credentials.clientSecret}`)}`);
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Review by jonkoops — APPROVED** [review]

**Comment by jonkoops** [comment]
Thanks for this @hmica, let's land these changes as they are. I've created #20368 to write the tests in the future.

**Review by jonkoops — CHANGES_REQUESTED** [review]
Looks like there are some linting issues that need to be resolved.

**Review by jonkoops — DISMISSED** [review]

**Review by jonkoops — APPROVED** [review]

**Pull request merged**